hw/net/hercules_emac.c

```diff
@@ -0,0 +1,650 @@
+#include <endian.h>
+#include "qemu/osdep.h"
+#include "hw/sysbus.h"
+#include "chardev/char-fe.h"
+#include "qemu/log.h"
+#include "qapi/error.h"
+#include "net/eth.h"
+#include "sysemu/dma.h"
+
+#include "hw/net/hercules_emac.h"
+
+typedef struct QEMU_PACKED HerculesCppiDescriptor {
+    uint32_t next;
+
+    uint32_t buffer_pointer;
+#ifdef HOST_WORDS_BIGENDIAN
```

Comment:
The only reason I could see why this is needed would be something to do with QEMUs DMA type, but even that would make me a little concerned. On the target, it seems like buffer_length should immediately follow the buffer pointer.

Reply:
As discussed, let's nuke it from the orbit as it were. We don't have a big endian host to validate it on and I'd rather not waste any time on use-case that currently have zero users.

Reply:
agreed.